wrappers/csharp/cs-tutorial-8-D400-on-chip-calibration/ExampleAutocalibrateDevice.cs

```diff
@@ -0,0 +1,541 @@
+                        {
+                            var aCalibratedDevice = AutoCalibratedDevice.FromDevice(dev);
+
+                            //calibratoin config set
+                            var calibrationSet = 0;
+                            foreach (var calibrationSpeed in GetEnumEntries<CalibrationSpeed>())
+                            {
+                                foreach (var calibrationScanParameter in GetEnumEntries<CalibrationScanParameter>())
+                                {
+                                    foreach (var calibrationDataSampling in GetEnumEntries<CalibrationDataSampling>())
+                                    {
+                                        if (calibrationDataSampling == CalibrationDataSampling.Windows && IsLinux)
+                                            continue;
+                                        Console.WriteLine("");
+#if (CalibrationTestAndBurns2FW)
+                                        // 1. Calibration table before running on-chip calibration. 
+                                        Console.WriteLine($"{Environment.NewLine}1. Calibration table before running on-chip calibration.");
+                                        var calTableBefore = aCalibratedDevice.CalibrationTable;
+                                        Console.WriteLine("\t Step 1 :  done");
+#endif
+                                        // 2. Runs the on-chip self-calibration routine that returns pointer to new calibration table.
+                                        Console.WriteLine($"2. Runs the on-chip self-calibration routine for depth profile {depthProfileSet} of {depthProfileSetCount} calibration, set {++calibrationSet} of {calibrationSetCount} ");
+                                        Console.WriteLine($"\t Format                    : {depthProfile.Format}");
+                                        Console.WriteLine($"\t Framerate                 : {depthProfile.Framerate}");
+                                        Console.WriteLine($"\t Width                     : {depthProfile.Width}");
+                                        Console.WriteLine($"\t Height                    : {depthProfile.Height}");
+                                        Console.WriteLine($"\t Calibration Speed         : {calibrationSpeed}");
+                                        Console.WriteLine($"\t Calibration Scan Parameter: {calibrationScanParameter}");
+                                        Console.WriteLine($"\t Calibration Data Sampling : {calibrationDataSampling}");
+
+                                        string calibrationConfig = GetCalibrationConfig(calibrationSpeed, calibrationScanParameter, calibrationDataSampling);
+
+                                        //var timeout = _timeoutForCalibrationSpeed[calibrationSpeed];
+                                        var timeout = Convert.ToInt32(TimeSpan.FromMinutes(2).TotalMilliseconds);
```

Comment:
Please add some comment to explain why 2 minutes are needed

Reply:
Not critical, the process may take about 20-30 sec. With USB2 this may take even longer